pkg/sns/topic.go

```diff
@@ -182,3 +212,66 @@ func (t *snsTopic) listSubscriptions() ([]*sns.Subscription, error) {
 
 	return subscriptions, nil
 }
+
+func (t *SnsTopic) subscriptionAttributesMatch(subscriptionArn *string, attributes map[string]interface{}) (bool, error) {
+	var ok bool
+	var err error
+	var subAttributes map[string]*string
+	var actualFilterPolicy *string
+	var expectedFilterPolicy []byte
+	var actualAttributes, expectedAttributes map[string]interface{}
+
+	if subAttributes, err = t.getSubscriptionAttributes(subscriptionArn); err != nil {
+		return false, err
+	}
+
+	if actualFilterPolicy, ok = subAttributes["FilterPolicy"]; !ok {
+		actualFilterPolicy = aws.String("null")
+	}
+
+	if err = json.Unmarshal([]byte(*actualFilterPolicy), &actualAttributes); err != nil {
+		return false, fmt.Errorf("can not unmarshal actual filter policy: %w", err)
+	}
+
+	if expectedFilterPolicy, err = json.Marshal(attributes); err != nil {
+		return false, fmt.Errorf("can not marshal expected filter policy: %w", err)
+	}
+
+	if err = json.Unmarshal(expectedFilterPolicy, &expectedAttributes); err != nil {
```

Comment:
Let me guess, this converts numbers in the map to float64 and similar?

Reply:
yes

Reply:
Would be nice to have an easier way to do that :thinking: Does mapstruct allow serialization from struct to struct? I know it allows weak typing